misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractCharSequenceToContainAtMostExpectationsTest.kt

```diff
@@ -0,0 +1,234 @@
+import ch.tutteli.atrium.specs.integration.CharSequenceToContainSpecBase.Companion.helloWorld
+import ch.tutteli.atrium.specs.integration.CharSequenceToContainSpecBase.Companion.toContainIgnoringCase
+
+abstract class AbstractCharSequenceToContainAtMostExpectationsTest(
+    private val toContainAtMostPair: Pair<(String, String) -> String, Fun3<CharSequence, Int, Any, Array<out Any>>>,
+    private val toContainAtMostIgnoringCasePair: Pair<(String, String) -> String, Fun3<CharSequence, Int, Any, Array<out Any>>>,
+    private val notToContainPair: Pair<String, (Int) -> String>,
+    private val exactlyPair: Pair<String, (Int) -> String>,
+) : ExpectationFunctionBaseTest() {
+
+    val toContainAtMostSpec = toContainAtMostPair.second
+    val toContainAtMostIgnoringCaseSpec = toContainAtMostIgnoringCasePair.second
+
+    @TestFactory
+    fun subjectLessTest() = subjectLessTestFactory(
+        toContainAtMostSpec.forSubjectLessTest(2, 2.3, arrayOf()),
+        toContainAtMostIgnoringCaseSpec.forSubjectLessTest(2, 2.3, arrayOf())
+    )
+
+    fun Expect<CharSequence>.toContainAtMostFun(atLeast: Int, a: Any, vararg aX: Any) =
+        toContainAtMostSpec(this, atLeast, a, aX)
+
+    fun Expect<CharSequence>.toContainAtMostIgnoringCaseFun(atLeast: Int, a: Any, vararg aX: Any) =
+        toContainAtMostIgnoringCaseSpec(this, atLeast, a, aX)
+
+    val valueWithIndent = "$indentRootBulletPoint$listBulletPoint$value"
+
+    @TestFactory
+    fun toContainAtMost__subject_throws_an_IllegalArgumentException() = testFactory(toContainAtMostSpec) { toContainAtMostFun ->
+        val (notToContain, errorMsgContainsNot) = notToContainPair
+        val (exactly, errorMsgExactly) = exactlyPair
+
+        it("for at most -1 -- only positive numbers") {
+            expect {
+                expect(text).toContainAtMostFun(-1, "")
+            }.toThrow<IllegalArgumentException> { messageToContain("positive number", -1) }
+        }
+        it("for at most 0 -- points to $notToContain") {
+            expect {
+                expect(text).toContainAtMostFun(0, "")
+            }.toThrow<IllegalArgumentException> { message { toEqual(errorMsgContainsNot(0)) } }
+        }
+        it("for at most 1 -- points to $exactly") {
+            expect {
+                expect(text).toContainAtMostFun(1, "")
+            }.toThrow<IllegalArgumentException> { message { toEqual(errorMsgExactly(1)) } }
+        }
+        it("if an object is passed as first expected") {
+            expect {
+                expect(text).toContainAtMostFun(2, expect(text))
+            }.toThrow<IllegalArgumentException> { messageToContain("CharSequence", "Number", "Char") }
+        }
+        it("if an object is passed as second expected") {
+            expect {
+                expect(text).toContainAtMostFun(2, "that's fine", expect(text))
+            }.toThrow<IllegalArgumentException> { messageToContain("CharSequence", "Number", "Char") }
+
```

Comment:
`}` is missing, please format the file, then it becomes obvious

Reply:
Maybe the line I chose for the review confused you, that's the line where the `}` is missing (not that `expect {` misses the ending `}`). The following will make it clearer:
![image](https://github.com/user-attachments/assets/42dd771f-eadb-40c4-8e9e-153ba5b05ec6)